### ISSUE TITLE
Remix Vite (v2.6.0) - `build/server` folder is not being generated alongside `build/client` folder after building app for production

### ISSUE THREAD

**Issue by cliffordfajardo** [remix-run/remix#8742]
### Background Context
I recently upgraded my remix app from esbuild compiler to use vite & I am using a custom server (express).
I am also using remix v1 routing using the v1 routes package  [@remix-run/v1-route-convention](https://github.com/remix-run/v1-compat-utils)

Everything is working fine during localhost development however, after I build my remix app for production (ex: `remix vite:build`) the `build` folder that gets generated does not include the `build/server` folder that should get generated along side `build/client`:

```sh
# missing server folder after `remix vite:build` is run
$ ls build 
client

# startup error logs
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/System/Volumes/Data/export/content/lid/apps/MY_APP/dev-i001/build/server/index.js'

```

<details>
<summary>FILE: vite.config.ts  </summary>

```js
export default defineConfig({
  plugins: [
    remix({
      // appDirectory: "app",
      // assetsBuildDirectory: "public/build",
      // serverBuildPath: "build/index.js",
      // publicPath: "/build/",
      serverModuleFormat: "esm",
      ignoredRouteFiles: ["**/.*"],
      routes(defineRoutes) {
        return Promise.resolve(createRoutesFromFolders(defineRoutes));
      },
    }),
    tsconfigPaths(),
  ],
  build: {
    target: "ES2022",
  },
  server: {
    https: {
      ca: HTTPS_CONFIG.ca,
      cert: HTTPS_CONFIG.cert,
      key: HTTPS_CONFIG.key,
      rejectUnauthorized: HTTPS_CONFIG.rejectUnauthorized,
      requestCert: HTTPS_CONFIG.requestCert,
    }
  },
});
```

</details>

<details>
<summary>FILE: tsconfig.json </summary>

```js
{
  "include": ["env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["**/*.test.*"],
  "compilerOptions": {
    "lib": ["DOM", "DOM.Iterable", "ES2022"],
    "isolatedModules": true,
    "esModuleInterop": true,
    "jsx": "react-jsx",
    "moduleResolution": "Bundler",
    "resolveJsonModule": true,
    "target": "ES2022",
    "module": "ES2022",
    "strict": true,
    "allowJs": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true,
    "baseUrl": ".",
    "paths": {
      "~/*": ["./app/*"],
    },

    // Vite takes care of building everything, not tsx
    "noEmit": true
  }
}

```

</details>

<details>
  <summary>FILE: package.json </summary>

```js
{
  "private": true,
  "sideEffects": false,
  "type": "module",
  "scripts": {
    "build": "cross-env NODE_ENV=production remix vite:build && yarn run typecheck",
    "start": "cross-env NODE_ENV=production NODE_TLS_REJECT_UNAUTHORIZED=0 tsx ./server/server.ts",
    // ... 
  }
}

```
</details>

## Reproduction / Sample Code
WORK IN PROGRESS
I have a hunch that maybe `@remix-run/v1-route-convention` might be influencing this because there are several v2 apps at work using vite 2.6.0 and the only difference is this app is using v1 routing with [@remix-run/v1-route-convention](https://github.com/remix-run/v1-compat-utils)

### System Info

```shell
System:
    OS: macOS 14.3.1
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 511.24 MB / 32.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.5.0 - ~/.volta/tools/image/node/20.5.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn
    npm: 9.8.0 - ~/.volta/tools/image/node/20.5.0/bin/npm
    bun: 1.0.3 - ~/.bun/bin/bun
  Browsers:
    Chrome: 121.0.6167.139
    Chrome Canary: 117.0.5849.2
    Safari: 17.3.1
  npmPackages:
    @remix-run/dev: 2.6.0 => 2.6.0 
    @remix-run/eslint-config: 2.6.0 => 2.6.0 
    @remix-run/express: 2.6.0 => 2.6.0 
    @remix-run/node: 2.6.0 => 2.6.0 
    @remix-run/react: 2.6.0 => 2.6.0 
    @remix-run/v1-route-convention: ^0.1.4 => 0.1.4 
    vite: ^4.5.0 => 4.5.2
```

### Used Package Manager

npm

### Expected Behavior

After running `remix vite:build`, the build/server folder is not getting generated

### Actual Behavior

After running `remix vite:build`, the build/server folder should always be generated since fullstack remix apps always have server code

**Comment by cliffordfajardo** [comment]
Might be related to #8744 - investigating 👀

**Comment by markdalgleish** [comment]
Was this the same `.DS_Store` issue as #8744?

**Comment by cliffordfajardo** [comment]
hi @markdalgleish - i found the issue!

✅  What solved my issue with remix 2.6.0 was upgrading from vite `^4.5.0` to `^5.0.0`

I verified this creating a new remix app with remix-vite express template:

1. Install fresh remix-express app
```
npx create-remix --template remix-run/remix/templates/unstable-vite-express
```
2. Change vite from version `^4.5.0` to `^5.0.0`
3. then build the app and you will see the `server` folder does not get generated when using vite pre version 5 on remix 2.6.0

![CleanShot 2024-02-13 at 14 27 44@2x](https://github.com/remix-run/remix/assets/6743796/e7c608c2-8820-403e-9484-d7e96cfaaf37)

**Comment by Shah-Sahab** [comment]
I am on vite 5.3.4 and still seeing this same issue on remix,

```
    "vite": "^5.3.4",
    "vite-tsconfig-paths": "^4.3.2"
```
```
    "@remix-run/node": "^2.5.1",
    "@remix-run/react": "^2.5.1",
    "@remix-run/serve": "^2.5.1",
```

 [vite] Error when evaluating SSR module virtual:remix/server-build: failed to import "@editorjs/editorjs"
|- ReferenceError: Element is not defined
